be/src/exec/json_parser.cpp

```diff
@@ -90,7 +90,9 @@ std::string JsonDocumentStreamParser::left_bytes_string(size_t sz) noexcept {
     if (off >= _len) {
         return {};
     }
-    return std::string(reinterpret_cast<char*>(_data) + off, _len - off);
+
+    auto len = std::min(_len - off, sz);
+    return std::string(reinterpret_cast<char*>(_data) + off, len);
```

Comment:
why not extract a left_bytes_string() for both parsers?

less code, more stable.

Reply:
`JsonArrayParser` and `JsonDocumentStreamParser` share the interface `JsonParser:left_bytes_string`.